### DISCUSSION TITLE
Access RTCPeerConnection of clients to achieve Audio streaming

### DISCUSSION THREAD

**Discussion by c4b4d4** [geckosio/geckos.io#159] [Help / Questions] [answered]
WebRTC is amazing for streaming video or audio, it's used on different frameworks to have solid voice-chat / radio rooms. Geckos.io uses WebRTC, which makes me thing voice-chatting would run good on it.

Is there any example on streaming data over Geckos.io? Has anyone come up with the feature I mentioned above.

I'm down to start developing this feature, but I think I need some guidence on where to start and how Geckos.io would fit on here.

Thanks in advance.

**Comment by c4b4d4** [comment]
I believe I have to get access to the ~~DataChannel~~ **RTCPeerConnection** opened by the dependency (https://www.npmjs.com/package/node-datachannel) ~~, which is under: `connection.channel.dataChannel`~~

**Update**

I've found I can access an instance of the PeerConnection with:  `channel.webrtcConnection.peerConnection` which belongs to the following class (https://github.com/geckosio/geckos.io/blob/61a801f504da04315d0985bbe8ef66f27afa0eea/packages/client/src/wrtc/peerConnection.ts):

```
export default class PeerConnection {
  dataChannel!: RTCDataChannel
  id: ChannelId
  localPeerConnection!: RTCPeerConnection
}
```

And here it is the **RTCPeerConnection** instance I need.

But whenever I do console log of `channel.webrtcConnection.peerConnection.id` I get `undefined` (Printing `channel.webrtcConnection.peerConnection` logs me `PeerConnection {}`

Seems that the class `PeerConnection ` doesn't ever get initiated or is not accesible how I want it.

I want to be able to:

`channel.webrtcConnection.peerConnection.localPeerConnection.addTrack();`

Accordingly to https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/addTrack

to be able to initiate the streaming of audio.

**Comment by yandeu** [comment, marked as answer]
node-datachannel can only handle client-server audio traffic.

What you are probably looking for is a client-to-client WebRTC library?

**Reply by c4b4d4** [reply]
I'm interested in client-server audio traffic. I want to stream from the server to the clients, kinda like a radio station.
